src/engine/QueryPlanner.cpp

```diff
@@ -1266,12 +1273,18 @@ void QueryPlanner::applyFiltersIfPossible(
               entityId = _qec->getIndex().getVocab().getValueIdForLT(compWith);
             } else if (filters[i]._type == SparqlFilter::LE) {
               entityId = _qec->getIndex().getVocab().getValueIdForLE(compWith);
+            } else if (filters[i]._type == SparqlFilter::LANG_MATCHES) {
+              entityId = std::numeric_limits<size_t>::max() - 1;
             }
           }
           std::shared_ptr<Operation> filter(
               new Filter(_qec, row[n]._qet, filters[i]._type,
```

Comment:
Use `std::make_shared`, also why is the right hand side string not set in the constructor like the others?

Reply:
Currently only the LANG_MATCHES filter requires an actual string as its right hand sight, all other types of filters work with an id. Due to that, to avoid adding an argument to the constructor that is not needed in most cases, the right hand side string is set separately right now.
A potentially better alternative solution would be adding a second constructor, that takes a string for the right hand side and then some assertions to assure that the right constructor is called for the right type of filter.